nf_core/components/update.py

```diff
@@ -299,10 +305,16 @@ def update(self, component=None, silent=False, updated=None, check_diff_exist=Tr
                 if recursive_update and len(modules_to_update + subworkflows_to_update) > 0:
                     # Update linked components
                     self.update_linked_components(modules_to_update, subworkflows_to_update, updated)
+                    self.manage_changes_in_linked_components(component, modules_to_update, subworkflows_to_update)
```

Comment:
can't we add `manage_changes_in_linked_components` inside `update_linked_components` because they should always be called together, no?

Reply:
We can't because of the recursion. We first update everything and then add/remove what we need. Otherwise, we are removing files that are updated afterwards.